### PR TITLE
Refine deepseekv2 modeling for to_static

### PR DESCRIPTION
**Pull request by zhangbo9674**
#### Before submitting

- [x] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [x] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### PR types

Bug fixes 
### PR changes

Others
### Description

Refine deepseekv2 modeling for to_static

### CODE REVIEW

File: paddlenlp/transformers/deepseek_v2/modeling_auto.py [hunk 1]
```diff
@@ -717,9 +716,6 @@ def forward(
             attention_mask = self._prepare_decoder_attention_mask(
```
Comment (DrownFish19) [inline paddlenlp/transformers/deepseek_v2/modeling_auto.py]: 如果attention_mask是支持use_cache的版本，那么就不是casual_mask，或者推理时使用left-padding，那么attention_mask也不是casual_mask，这里的修改不能覆盖之前的情况

Comment (zhangbo9674) [reply paddlenlp/transformers/deepseek_v2/modeling_auto.py]: 这里的改动，主要是因为动转静不支持如下场景：
```
if self.config.use_flash_attention:
    attention_mask = None if is_casual_mask(attention_mask) else attention_mask
```
当前这样该是基于目前的场景下 use_flash_attention 下 attention_mask 一定为 None，还有一种改法，就是将控制流判断后移到调用的地方，但是目前这种情况下，自动并行的切分推导对控制流的场景支持还存在一些问题，因此为了不影响后续流程，先按照前面的改法实现
```
if is_casual_mask(attention_mask):
    layer_outputs = decoder_layer(
                hidden_states=hidden_states,
                position_ids=position_ids,
                attention_mask=None,
                output_attentions=output_attentions,
                past_key_value=past_key_value,
                use_cache=use_cache,
                attn_mask_startend_row_indices=attn_mask_startend_row_indices)
else:
    layer_outputs = decoder_layer(
                hidden_states=hidden_states,
                position_ids=position_ids,
                attention_mask=attention_mask,
                output_attentions=output_attentions,
                past_key_value=past_key_value,
                use_cache=use_cache,
                attn_mask_startend_row_indices=attn_mask_startend_row_indices)
```

**Review by DrownFish19 — DISMISSED** [review]
LGTM

File: paddlenlp/transformers/deepseek_v2/modeling.py [hunk 2]
```diff
@@ -231,11 +231,8 @@ def scaled_dot_product_attention(
             )
 
         attn_weights = attn_weights + attention_mask
-        if not paddle.in_dynamic_mode():
+        with paddle.amp.auto_cast(False):
```
Comment (DrownFish19) [inline paddlenlp/transformers/deepseek_v2/modeling.py:234]: 这里修改的原因是？

**Review by DrownFish19 — APPROVED** [review]
LGTM
